tools/loader/ModelCompiler.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Loader.h"
+
+#include "glow/Converter/TypeAToTypeBFunctionConverter.h"
+#include "glow/Graph/Nodes.h"
+#include "glow/Importer/Caffe2ModelLoader.h"
+#include "glow/Importer/ONNXModelLoader.h"
+
+#include "llvm/Support/CommandLine.h"
+
+using namespace glow;
+
+namespace {
+
+/// Model compiler options.
+llvm::cl::OptionCategory modelCompilerCat("Model Compiler Options");
+
+llvm::cl::list<std::string> modelInputs(
+    "model-input", llvm::cl::ZeroOrMore,
+    llvm::cl::desc(
+        "For ONNX models the inputs of the graph can be inferred   \n"
+        "automatically and hence this option is not mandatory.     \n"
+        "For Caffe2 models the graph definition does not contain   \n"
+        "the description of the inputs and hence must be provided  \n"
+        "explicitly using this option. One or more model inputs    \n"
+        "are provided using the following format:                  \n"
+        "   -model-input=<inputName1>,<inputType1>,<inputShape1>   \n"
+        "   -model-input=<inputName2>,<inputType2>,<inputShape2>   \n"
+        "   ....................................................   \n"
+        "For quantized types the format is slightly different since\n"
+        "the scale and offset parameters should also be provided:  \n"
+        "   -model-input=<name>,<type>,<scale>,<offset>,<shape>    \n"
+        "For example we can can provide one or more inputs:        \n"
+        "   -model-input=input_03_data,float,[1]                   \n"
+        "   -model-input=data_bias,int32,[1,32,32]                 \n"
+        "   -model-input=data,int8q,0.123,-13,[1,10]               \n"
+        "The supported types are:                                  \n"
+        "   - float, float16 (floating point types)                \n"
+        "   - int32, int64 (integer types)                         \n"
+        "   - int8q, int16q, int32q (integer quantized types)      \n"
+        "   - bool (logic type)\n"),
+    llvm::cl::value_desc("name,type,shape"), llvm::cl::cat(modelCompilerCat));
+
+} // namespace
+
+/// Parse the strings and get the model input names and types.
```

Comment:
Please add in the comment the description of the syntax being parsed by this function.

Reply:
Done.